internal/http/manage_runner_health.go

```diff
@@ -67,24 +90,47 @@ func MonitorRunnerHealth(ctx context.Context, cmd *exec.Cmd, runnerServerURI str
 			select {
 			case <-ctx.Done():
 				logs.Debug("Stopped monitoring runner health")
+				resultChan <- healthCheckResult{Status: StatusMonitoringCancelled}
 				return
+
 			case <-ticker.C:
 				if err := sendRunnerHealthCheckRequest(runnerServerURI); err != nil {
 					failureCount++
 					logs.Warnf("Found runner unresponsive (%d/%d)", failureCount, healthCheckMaxFailures)
 					if failureCount >= healthCheckMaxFailures {
-						logs.Warn("Reached max failures on runner health check, terminating runner...")
-						if err := cmd.Process.Kill(); err != nil {
-							panic(fmt.Errorf("failed to terminate runner process: %v", err))
-						}
-						logs.Debug("Stopped monitoring runner health")
+						logs.Warn("Found runner unresponsive too many times, terminating runner...")
```

Comment:
Since this no longer does the termination, we could move this log to where it's actually terminated